packages/cli/src/ui/state/extensions.ts

```diff
@@ -83,6 +95,27 @@ export function extensionUpdatesReducer(
         ...state,
         batchChecksInProgress: state.batchChecksInProgress - 1,
       };
+    case 'SCHEDULE_UPDATE':
+      return {
+        ...state,
+        // If there is a pre-existing scheduled update, we merge them.
+        scheduledUpdate: {
+          all: state.scheduledUpdate?.all || action.payload.all,
+          names: [
+            ...(state.scheduledUpdate?.names ?? []),
+            ...(action.payload.names ?? []),
+          ],
+          onComplete: (updateInfos) => {
+            state.scheduledUpdate?.onComplete(updateInfos);
```

Comment:
this is a bit complex. instead of chaining this together can you make it explicit that you have a list of onComplete methods that each need to be called? you can then do a for loop calling all of them when you get outside the reducer keeping the callstacks from getting oddly chained.
also add try catch blocks so if one onComplete throws the others still occur.

Reply:
done